### PR TITLE
Add logout options

### LINKED ISSUE #842: Support federated params for logout
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/nextjs-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

Please add support for https://auth0.com/docs/authenticate/login/logout/log-users-out-of-idps

### Describe the ideal solution

Add federated as part of the query params in https://github.com/auth0/nextjs-auth0/blob/main/src/auth0-session/client.ts#L121

### Alternatives and current workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by adamjmcgrath**
### 📋 Changes

For example, to pass the federated logout param per https://auth0.com/docs/authenticate/login/logout/log-users-out-of-idps

```js
handleLogout(req, res, { logoutParams: { federated: '' } });
```

### 📎 References

fixes #842

### CODE REVIEW

File: src/auth0-session/config.ts [hunk 1]
```diff
@@ -287,4 +287,15 @@ export interface LogoutOptions {
...
    */
   returnTo?: string;
+
+  /**
+   * Additional custom parameters to pass to the logout endpoint.
+   *
+   * @example pass the federated logout param per https://auth0.com/docs/authenticate/login/logout/log-users-out-of-idps
+   *
+   * ```js
+   * handleLogout(req, res, { logoutParams: { federated: '' } });
```
Comment (Widcket) [inline src/auth0-session/config.ts:297]: Shouldn't it be `federated: true` instead of an empty string?

**Review by Widcket — COMMENTED** [review]
Just have a question about the `federated` API.

File: src/auth0-session/config.ts (hunk 1 above)
Comment (adamjmcgrath) [reply src/auth0-session/config.ts:297]: `federated=true` would work, but Auth0 asks for `https://YOUR_DOMAIN/v2/logout?federated` - [in its docs](https://auth0.com/docs/authenticate/login/logout/log-users-out-of-idps#:~:text=https%3A//YOUR_DOMAIN/v2/logout%3Ffederated) and this is a closer approximation (which also works)

Comment (adamjmcgrath) [reply src/auth0-session/config.ts:297]:
```bash
$ node -e "foo=new URL('https://YOUR_DOMAIN/v2/logout'); foo.searchParams.set('federated', ''); console.log(foo.toString())"
https://your_domain/v2/logout?federated=
```

**Review by Widcket — APPROVED** [review]

**Pull request merged**